src/Control/Monad/Aff.js

```diff
@@ -293,3 +293,70 @@ exports._liftEff = function (nonCanceler, e) {
     return nonCanceler;
   };
 }
+
+exports._tailRecM = function (isLeft, f, a) {
+  return function(success, error) {
+    return function go(acc) {
+      var result, status, canceler;
+
+      // Observes synchronous effects using a flag.
+      //
+      //   status = 0 (unresolved status)
+      //   status = 1 (synchronous effect)
+      //   status = 2 (asynchronous effect)
+      //
+      // The inner Aff is run, and if the status is still 0, we have observed
+      // a synchronous effect. The result is stashed for later synchronous
```

Comment:
Can you explain why the status being still `0` means we've observed a synchronous effect? I'm sure you're right, but since I don't really know the guts of Aff too well, it wasn't quite clear to me how.